src/new/wallet.js

```diff
@@ -2666,6 +2675,168 @@ class HathorWallet extends EventEmitter {
+   * @param {any[]} [data.args] List of arguments for the method to be executed in the transaction
+   * @param [options]
+   * @param {string} [options.pinCode] PIN to decrypt the private key.
+   *                                   Optional but required if not set in this
+   *
+   * @returns {Promise<NanoContract>}
+   */
+  async createAndSendNanoContractTransaction(method, address, data, options = {}) {
+    if (await this.storage.isReadonly()) {
+      throw new WalletFromXPubGuard('createAndSendNanoContractTransaction');
+    }
+    const newOptions = Object.assign({ pinCode: null }, options);
+    const pin = newOptions.pinCode || this.pinCode;
+    if (!pin) {
+      throw new PinRequiredError(ERROR_MESSAGE_PIN_REQUIRED);
+    }
+
+    if (method !== 'initialize' && !data.ncId) {
+      // Initialize is the only method that creates the nano contract
+      // the other methods must have the corresponding ncId to be executed
+      throw new NanoContractTransactionError('Missing nano contract id. Parameter ncId in data');
+    }
+
+    if (method === 'initialize' && !data.blueprintId) {
+      // Initialize needs the blueprint ID
+      throw new NanoContractTransactionError('Missing blueprint id. Parameter blueprintId in data');
+    }
+
+    let blueprintId = data.blueprintId;
+
+    if (method !== 'initialize') {
+      let response;
+      try {
+        response = await this.getFullTxById(data.ncId);
+      } catch {
+        // Error getting nano contract transaction data from the full node
+        throw new NanoContractTransactionError(`Error getting nano contract transaction data with id ${data.ncId} from the full node`);
+      }
+
+      if (response.tx.version !== NANO_CONTRACTS_VERSION) {
+        throw new NanoContractTransactionError(`Transaction with id ${data.ncId} is not a nano contract transaction.`);
+      }
+
+      blueprintId = response.tx.nc_blueprint_id;
+    }
+
+    // Get the blueprint data from full node
+    const blueprintInformation = await ncApi.getBlueprintInformation(blueprintId);
+    const methodArgs = get(blueprintInformation, `public_methods.${method}.args`);
+    if (!methodArgs) {
+      throw new NanoContractTransactionError(`Blueprint does not have method ${method}.`);
+    }
+
+    // Args may come as undefined
+    const argsLen = data.args ? data.args.length : 0;
+    if (argsLen !== methodArgs.length) {
+      throw new NanoContractTransactionError(`Method needs ${methodArgs.length} parameters but data has ${data.args.length}.`);
+    }
+
+    // Here we validate that the arguments sent in the data array of args has
+    // the expected type for each parameter of the blueprint method
+    for (const [index, arg] of methodArgs.entries()) {
+      let typeToCheck = arg.type;
+      if (typeToCheck.startsWith('SignedData')) {
+        // Signed data will always be an hexadecimal with the
+        // signed part and the data itself
```

Comment:
Isn't it 3 parts?
len + signature + data

Reply:
Yes, when I wrote 'signed part' it was including the len of signature + signature. But I agree that it's better to detail more. https://github.com/HathorNetwork/hathor-wallet-lib/pull/604/commits/315ce908baf61805a51d9c69f3d0a89d324ffded